WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift

```diff
@@ -130,8 +128,173 @@ final class ProductFormViewController: UIViewController {
 
         view.endEditing(true)
     }
+
+    // MARK: - Navigation actions handling
+
+    override func shouldPopOnBackButton() -> Bool {
+        if viewModel.hasUnsavedChanges() {
+            presentBackNavigationActionSheet()
+            return false
```

Comment:
What do you think about using `guard let` to force the return statement and to  be aligned with code added here https://github.com/woocommerce/woocommerce-ios/pull/2565/files#diff-2929b8aeca63d242a89ab243c9f7ddacR167

Reply:
the code link doesn't really show related code on my browser, but I like the idea of using `guard let` to force the return statement. updated here: https://github.com/woocommerce/woocommerce-ios/pull/2565/commits/55a1f13fa8573d4bf66a0359e920e2acaf78d97f